osu.Game/Overlays/BeatmapSet/Scores/ScoreTable.cs

```diff
@@ -65,6 +66,10 @@ public IReadOnlyList<ScoreInfo> Scores
                 for (int i = 0; i < value.Count; i++)
                     backgroundFlow.Add(new ScoreTableRowBackground(i, value[i]));
 
+                // Ensure correct column order
+                foreach (ScoreInfo score in value)
+                    score.Statistics = score.Statistics.OrderByDescending(pair => pair.Key).ToDictionary(pair => pair.Key, pair => pair.Value);
```

Comment:
Shouldn't be mutating `ScoreInfo` IMO. Would be best if `ScoreInfo` exposed a read-only ordered view of rank-count pairs.

Reply:
That would indeed be a good way to do it. 

I think I'll split this as well as the `FormatAccuracy` changes into separate PRs, since their scope is beyond just the beatmapset overlay.